### PR TITLE
Holdings extended by Dividend Yield Percentage

### PR DESCRIPTION
**Pull request by Fullstack-Arshad**
Feature: #2533

### CODE REVIEW

**Comment by Fullstack-Arshad** [comment]
Hi @dtslvr, I have used overall dividend to calculate dividend percentage, let me know if any other change is required :)

**Comment by Fullstack-Arshad** [comment]
do you want me to add ok-dividend json file as well

**Comment by Fullstack-Arshad** [comment]
Hi @dtslvr , is there any other change require? Let me know please

File: apps/api/src/app/portfolio/portfolio.service.ts [hunk 1]
```diff
@@ -836,6 +837,9 @@ export class PortfolioService {
           })
       );
 
+      const dividendYieldInPercent =
+        (dividendInBaseCurrency.toNumber() * 100) / marketPrice;
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/portfolio.service.ts:841]: Can you refactor the calculation using `Big.js`?

```
dividendInBaseCurrency.mul(100).div(marketPrice);
```

File: apps/api/src/app/portfolio/portfolio.service.ts [hunk 2]
```diff
@@ -930,6 +934,7 @@ export class PortfolioService {
         averagePrice: averagePrice.toNumber(),
         dataProviderInfo: portfolioCalculator.getDataProviderInfos()?.[0],
         dividendInBaseCurrency: dividendInBaseCurrency.toNumber(),
+        dividendYieldInPercent: dividendYieldInPercent,
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/portfolio.service.ts:937]: `dividendYieldInPercent: dividendYieldInPercent.toNumber();`

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for your contribution, @Fullstack-Arshad! 🙂
I have added a few comments.

File: apps/api/src/app/portfolio/portfolio.service.ts (hunk 2 above)
Comment (Fullstack-Arshad) [reply apps/api/src/app/portfolio/portfolio.service.ts]: done. Please Check

**Comment by Joren-vanGoethem** [comment]
this is a feature i would really like to see, should we fork it and re-open the merge so that we can actually merge it since it seems like he is inactive on github now...

**Comment by Joren-vanGoethem** [comment]
> I am there, is there still something pending? @dtslvr said code is ready to merge now, I was under the impression that it is completed. I can continue if anything is required.

you need to actually initiate the merge i think, unless @dtslvr can do it

**Comment by dtslvr** [comment]
> you need to actually initiate the merge i think

This PR is fine from a technical point of view. Currently, there are a few things changing in the performance calculation, so I wanted to wait with the integration. In the meantime, I have realized that the calculation with all-time dividend is probably too simple and needs to be annualized. What do you think?

**Comment by Fullstack-Arshad** [comment]
Yes I felt the same while Implementing this

**Comment by dtslvr** [comment]
> Yes I felt the same while Implementing this

There is a `getAnnualizedPerformancePercent()` function in [PortfolioCalculator](https://github.com/ghostfolio/ghostfolio/blob/main/apps/api/src/app/portfolio/portfolio-calculator.ts#L156). Maybe that would be enough?

**Comment by Fullstack-Arshad** [comment]
I am sorry I didn't get you @dtslvr, in order to calculate dividend yield percentage I need annualized dividend, How can I get annualized dividend using getAnnualizedPerformancePercent(). Can you please explain bit more.

**Comment by dtslvr** [comment]
You currently calculate the (total) dividend yield with the sum of all dividend. My idea is to annualize (normalization) this dividend yield with the number of days you are holding this position (via date of first activity) using `getAnnualizedPerformancePercent()`.

**Pull request merged**